scripts/ci/check_compliance.py

```diff
@@ -323,6 +323,8 @@ def run(self):
             if len(matches) == 0:
                 self.failure(output)
 
+        diff.communicate()
```

Comment:
I will let @pdgendt comment and confirm but I think the "Python way" would be to use a context manager

```py
with subprocess.Popen(.. blah... ) as diff:
	try:
		...
...
```

Reply:
Preferably, yes